eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java

```diff
@@ -37,19 +31,22 @@
     /** The instance. */
     static CodeCheckerContext instance;
 
-    /** The job runner. */
-    // TODO UPLIFT jobrunner will be a thread.
-    //JobRunner jobRunner;
+    private static final String NULL_WINDOW = "Error: Active Workbench Window is null!";
 
     /** The active editor part. */
     IEditorPart activeEditorPart;
 
-    /** The servers. */
-    private HashMap<IProject, CodecheckerServerThread> servers = new HashMap<>();
+    /** The active project. */
+    private IProject activeProject;
 
-    private HashMap<IProject, SearchList> reports = new HashMap<>();
+    private Map<IProject, SearchList> reports = new HashMap<>();
```

Comment:
Does this not cause problems if the type is `Map` in the `new HashMap` is used?

Reply:
I think it's better to use the interfaces at definition, parameter passing, etc rather than the exact implementation.  The original code is full of restrictions like this.